Source/WebCore/Modules/identity/DigitalCredential.cpp

```diff
@@ -47,6 +63,60 @@ DigitalCredential::DigitalCredential(JSC::Strong<JSC::JSObject>&& data, Identity
 {
 }
 
+void DigitalCredential::discoverFromExternalSource(const Document& document, CredentialPromise&& promise, CredentialRequestOptions&& options)
+{
+    ASSERT(options.digital);
+
+    if (options.mediation != MediationRequirement::Required) {
+        promise.reject(Exception { ExceptionCode::TypeError, "User mediation is required for DigitalCredential."_s });
+        return;
+    }
+
+    if (!PermissionsPolicy::isFeatureEnabled(PermissionsPolicy::Feature::DigitalCredentialsGetRule, document, PermissionsPolicy::ShouldReportViolation::No)) {
+        promise.reject(Exception { ExceptionCode::NotAllowedError, "Third-party iframes are not allowed to call .get() unless explicitly allowed via Permissions Policy (digital-credentials-get)"_s });
+        return;
+    }
+
+    if (!document.protectedFrame() || !document.protectedFrame()->protectedPage() || !document.protectedWindow()) {
+        LOG(DigitalCredentials, "Preconditions for DigitalCredential.get() are not met");
+        promise.reject(ExceptionCode::InvalidStateError, "Preconditions for calling .get() are not met."_s);
+        return;
+    }
+
+    if (!document.hasFocus()) {
+        promise.reject(Exception { ExceptionCode::NotAllowedError, "The document is not focused."_s });
+        return;
+    }
+
+    if (document.visibilityState() != VisibilityState::Visible) {
+        promise.reject(Exception { ExceptionCode::NotAllowedError, "The document is not visible."_s });
+        return;
+    }
+
+    if (options.digital->requests.isEmpty()) {
+        promise.reject(Exception { ExceptionCode::TypeError, "At least one request must present."_s });
+        return;
+    }
+
+    if (!document.protectedWindow()->consumeTransientActivation()) {
+        promise.reject(Exception { ExceptionCode::NotAllowedError, "Calling get() needs to be triggered by an activation triggering user event."_s });
+        return;
+    }
+
+#if HAVE(DIGITAL_CREDENTIALS_UI)
+    DigitalCredentialsRequestData requestData;
+    requestData.options = options.digital.value();
+    requestData.topOrigin = document.protectedTopOrigin()->data().isolatedCopy();
+    requestData.documentOrigin = document.protectedSecurityOrigin()->data().isolatedCopy();
+
+    Ref coordinator = document.frame()->page()->credentialRequestCoordinator();
```

Comment:
shouldn't you test that frame() or page() aren't null?

Reply:
Tested them on line 80, but will switch to protectedFrame()->protectedPage()